===================================================================
RCS file: /repo/OpenLDAP/pkg/ldap/servers/slapd/bconfig.c,v
retrieving revision 1.145
retrieving revision 1.147
diff -u -r1.145 -r1.147
--- servers/slapd/bconfig.c	2005/10/05 18:07:32	1.145
+++ servers/slapd/bconfig.c	2005/10/09 19:20:40	1.147
@@ -3437,6 +3437,11 @@
 		BackendDB *be = op->o_bd;
 		slap_callback sc = { NULL, slap_null_cb, NULL, NULL };
 		op->o_bd = &cfb->cb_db;
+		/* FIXME: there must be a better way. */
+		if ( ber_bvcmp( &op->o_bd->be_rootndn, &be->be_rootndn )) {
+			op->o_bd->be_rootdn = be->be_rootdn;
+			op->o_bd->be_rootndn= be->be_rootndn;
+		}
 		sc.sc_next = op->o_callback;
 		op->o_callback = &sc;
 		op->o_bd->be_add( op, rs );
@@ -3761,6 +3766,10 @@
 		BackendDB *be = op->o_bd;
 		slap_callback sc = { NULL, slap_null_cb, NULL, NULL };
 		op->o_bd = &cfb->cb_db;
+		if ( ber_bvcmp( &op->o_bd->be_rootndn, &be->be_rootndn )) {
+			op->o_bd->be_rootdn = be->be_rootdn;
+			op->o_bd->be_rootndn= be->be_rootndn;
+		}
 		sc.sc_next = op->o_callback;
 		op->o_callback = &sc;
 		op->o_bd->be_modify( op, rs );
